init/eessi_archdetect.sh

```diff
@@ -150,8 +150,25 @@ cpupath(){
     fi
 }
 
+accelpath() {
+    # If EESSI_ACCELERATOR_TARGET_OVERRIDE is set, use it
+    log "DEBUG" "accelpath: Override variable set as '$EESSI_ACCELERATOR_TARGET_OVERRIDE' "
+    [ $EESSI_ACCELERATOR_TARGET_OVERRIDE ] && echo ${EESSI_ACCELERATOR_TARGET_OVERRIDE} && exit
+
+    # check for NVIDIA GPUs via nvidia-smi command
+    nvidia_smi=$(command -v nvidia-smi)
+    if [[ $? -eq 0 ]]; then
+        log "DEBUG" "accelpath: nvidia-smi command found @ ${nvidia_smi}"
+	gpu_info=$(nvidia-smi --query-gpu=gpu_name,count,driver_version,compute_cap --format=csv,noheader | head -1)
+	cuda_cc=$(echo $gpu_info | sed 's/, /,/g' | cut -f4 -d, | sed 's/\.//g')
+	echo "accel/nvidia/cc${cuda_cc}"
+    else
+        log "DEBUG" "accelpath: nvidia-smi command not found"
```

Comment:
Is this enough? It seems very silent, no error code, just no value returned

Reply:
I'm not sure what else you would want here... `command -v nvidia-smi` didn't find `nvidia-smi`, so game over?

Should we print value of `$PATH`?

Reply:
What I mean is, shouldn't this be an error? They asked for `accelpath` but don't have the tools needed to figure it out (nor the override set). Shouldn't we exit with an error (and a helpful message)?

Reply:
If we exit with an error here, then how do we deal with that in the `init` script (or `EESSI` module)?

There's no guarantee there will be an accelerator, so returning nothing is a good way of handling that common case?

Reply:
We just allow for a possible error in the init script? We could use the string returned if the command succeeds, and we don't if the command errors. For CPU a failure is catastrophic but for accel we can manage it.

If we think of `archdetect` as a standalone utility this is a bit important for usability.

Reply:
You can argue both ways I guess: since not having a GPU is a common case, I didn't want to make it an error.

But your point about archdetect also being a standalone utility is a good one for making it a non-zero exit if no GPU could be detected, and then dealing with that correctly in our init script.

I'll adopt things accordingly...